pandas/tests/indexing/test_interval.py

```diff
@@ -39,6 +42,9 @@ def test_loc_with_scalar(self):
         expected = s.iloc[2:5]
         tm.assert_series_equal(expected, s.loc[s >= 2])
 
+        # test endpoint of non-overlapping monotonic decreasing (GH16417)
```

Comment:
I would instead make a specific test of this (also test increasing monotonic), even better would be to parametrize over closed (bonus points)